rayon-core/src/lib.rs

```diff
@@ -170,6 +173,12 @@ type StartHandler = Fn(usize) + Send + Sync;
 /// Note that this same closure may be invoked multiple times in parallel.
 type ExitHandler = Fn(usize) + Send + Sync;
 
+/// The type for a closure that gets invoked with a
+/// function which runs rayon tasks.
+/// The closure is passed the index of the thread on which it is invoked.
+/// Note that this same closure may be invoked multiple times in parallel.
```

Comment:
How is this distinct from the `StartHandler`?

Reply:
Oh, I see, it kind of "takes over" the thread

Reply:
It allows up to run code both before and after the Rayon worker, so it is more powerful than `StartHandler` and `ExitHandler`. [You can see it used here](https://github.com/rust-lang/rust/pull/50235/files#diff-698522bf32a92da4331b4bd3f6ed31c6R115).